### PR TITLE
Fixes for custom dataloaders

### PR DESCRIPTION
**Pull request by jdunnmon**

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -15,3 +15,6 @@ dependencies:  # New core deps should also be added to setup.py for pip
   - jupyter
   - nb_conda_kernels
   - runipy
+  - pip
+  - pip:
+    - gputil
```
Comment (ajratner) [inline environment.yml]: I still think we shouldn't need this- tests were passing on master without this

File: metal/end_model/end_model.py [hunk 2]
```diff
@@ -111,7 +111,8 @@ def _make_layer(self, module, output_dim=None):
         if isinstance(module, IdentityModule):
             return module
         layer = [module]
-        layer.append(nn.ReLU())
+        if self.config["relu"]:
+            layer.append(nn.ReLU())
```
Comment (ajratner) [inline metal/end_model/end_model.py]: I think the cleaner fix is to deal with the no-middle layers case properly, as we discussed; I'll work on that

File: metal/tuners/random_tuner.py [hunk 3]
```diff
@@ -66,7 +65,22 @@ def search(
             # Unless seeds are given explicitly, give each config a unique one
             if config.get("seed", None) is None:
                 config["seed"] = self.seed + i
-            model = self.model_class(*init_args, **init_kwargs, **config)
+
+            # Integrating generated config into init kwargs and train kwargs
+            init_kwargs = recursive_merge_dicts(init_kwargs, config)
+            train_kwargs = recursive_merge_dicts(train_kwargs, config)
+
+            # Removing potential duplicate keys -- error occurs without this!
```
Comment (ajratner) [inline metal/tuners/random_tuner.py]: ??

File: metal/tuners/tuner.py [hunk 4]
```diff
@@ -77,6 +75,12 @@ def get_run_stats(self):
         """
         return self.run_stats
 
+    def remove_key(self, d, key):
```
Comment (ajratner) [inline metal/tuners/tuner.py]: Delete

File: setup.py [hunk 5]
```diff
@@ -50,6 +50,7 @@ def run(self):
         "torch>=0.4.1",
         "scipy",
         "tqdm",
+        "GPUtil",
```
Comment (ajratner) [inline setup.py]: Remove

File: metal/tuners/random_tuner.py (hunk 3 above)
Comment (jdunnmon) [reply metal/tuners/random_tuner.py]: To make sure we don't have duplicate keys when we're doing model search -- if init_kwargs and config have duplicate keys, we error out.

**Pull request merged**